codalab/model/bundle_model.py

```diff
@@ -1329,13 +1333,13 @@ def add_worksheet_items(self, worksheet_uuid, items, after_sort_key=None, replac
                             'subworksheet_uuid': item.subworksheet_uuid,
                             'value': item.value,
                             'type': item.type,
-                            'sort_key': item_sort_key(item) * 2 + offset,
+                            'sort_key': item_sort_key(item) + offset,
```

Comment:
Was this a bug?  Can you explain what goes wrong / how things are fixed?

Reply:
This line of code moves the items _after_ the location where you're inserting down by `offset`, by adding `offset` to the sort key. However, the previous code was first doubling the sort key and then adding `offset` to it.

I think the new code gets the logic right; not sure why one would want to double the sort key as in the old code. I'm not exactly sure how doubling it led to the completely unexpected behavior documented in #1462, though.

Reply:
Looking from the if clause in the code, 1323 (in commit) or 1319(in original), 
the condition min(item_sort_key(item) for item in after_items) - after_sort_key <= offset indicates that the distance between the sort_key of the original after_items and after_sort_key is less than offset, which means there aren't enough integer for sort_keys between min(all after_items_sort_key) and after_sort_key, I think that's the original reason why the code doubled item_sort_key, which seems unreasonable

Reply:
I feel like it's not quite right to just change the sort_key to sort_key + offset, since the if condition means sort_key + offset < after_sort_key + len(items), which sounds like the new sort key would overlap with some of the inserted items' sort key, not completely sure

Reply:
Actually my original thoughts were wrong, if we look at the error comment example in the issue #1462 we see  : 

{"items":["## **My running bundles**","These are bundles that are currently running or queued to be run.","a","b","d","c",""],"after_sort_key":-28,"ids":[676,677]}

sort keys can be assigned negative numbers, which is why *2 can mess up the ordering, because we place smaller sort_key in front and larger sort_key on the back, if the sort key was assigned negative number, *2 makes it smaller and move it to the front and thus the mess up.

Example: 
row 1: key -27
row 2: key -28
we want to insert an item between row 1 & row 2, the original code will cause the row 2 key be -56 + 1 = -55 and thus will be placed before row 1 after insertion

So the original code was a bug and removing *2 is good.

Reply:
Great explanation!